test/ectest.c

```diff
@@ -1820,7 +1820,7 @@ static int parameter_test(void)
     unsigned char *buf = NULL;
     int r = 0, len;
 
-    if (!TEST_ptr(group = EC_GROUP_new_by_curve_name(NID_secp112r1))
+    if (!TEST_ptr(group = EC_GROUP_new_by_curve_name(NID_secp224r1))
```

Comment:
@romen This subtle change has introduced a test failure if I configure with this line 
./config --debug no-asm enable-ec_nistp_64_gcc_128
then a test fails...
../test/recipes/15-test_ec.t ....................... 
1..5
ok 1 - require '../../test/recipes/tconversion.pl';
    # Subtest: ../../test/ectest
    1..12
    # ERROR: (int) 'EC_GROUP_cmp(group, group2, NULL) == 0' failed @ test/ectest.c:1826
    # [1] compared to [0]
    not ok 1 - parameter_test
    # 00:47:EC:93:A4:7F:00:00:error:10071065:elliptic curve routines:EC_POINT_cmp:incompatible objects:crypto/ec/ec_lib.c:890:
    # INFO:  @ test/ectest.c:184
    # Curve defined by Weierstrass equation
    #      y^2 = x^3 + a*x + b (mod p)


This is because the EC_GROUP_new_by_curve_name  uses the EC_GFp_nistp224_method
and the EC_GROUP_new_curve_GFp uses EC_GFp_nist_method (or potentially EC_GFp_mont_method).

Reply:
Yes, I noticed this yesterday as well: I am opening an issue to discuss it because I don't see an easy fix, given that it was always tested only on a single curve which did not have multiple ec_methods that test has actually been failing for way longer than the pr that made it surface.

Reply:
See #9251